### PR TITLE
Add dockerfiles

### PR DESCRIPTION
**Pull request by mmorhun**
Signed-off-by: Mykola Morhun <mmorhun@redhat.com>

### CODE REVIEW

File: .github/workflows/pr-check.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+# which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+name: PR check
+on:
+  pull_request:
+jobs:
+  build:
+    runs-on: ubuntu-latest
```
Comment (tolusha) [inline .github/workflows/pr-check.yaml]: It is better not to use latest but a specific version.

File: .github/workflows/build-publish.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      - name: Docker login
+        uses: azure/docker-login@v1
+        with:
+          login-server: quay.io
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      - name: Build and push image
+        run: |
+          TAG=$(git rev-parse --short HEAD)
+          docker build -t ${IMAGE_NAME}:${TAG} .
```
Comment (tolusha) [inline .github/workflows/build-publish.yaml]: Are we going to provide multi-platform build?

**Comment by tolusha** [comment]
Having README.md explaining the purpose of the repository will be nice.

File: .github/workflows/pr-check.yaml (hunk 1 above)
Comment (mmorhun) [reply .github/workflows/pr-check.yaml]: In most cases latest is used, but it is easy to change. Will do.

File: .github/workflows/build-publish.yaml (hunk 2 above)
Comment (mmorhun) [reply .github/workflows/build-publish.yaml]: Ok, I will

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+# This program and the accompanying materials are made
+# available under the terms of the Eclipse Public License 2.0
+# which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+
+FROM alpine:3.13.4
+
+ARG ARCH=amd64
```
Comment (tolusha) [inline Dockerfile]: Can we detect the architecture ?

File: README.md [hunk 4]
```diff
@@ -2,4 +2,6 @@
 
 This repository contains internal REST backup server image dockerfiles for Eclipse Che.
 
+The image uses restic REST server that supposed to store Che backups.
+
 It is designed to be deployed by Che Operator by setting `autoconfigureRestBackupServer` to `true` in an instance of `CheClusterBackup` and be accecible only from within the cluster.
```
Comment (tolusha) [inline README.md]:
```suggestion
```

Can remove this? Because if we change the field name we will have to update the README.
Instead of this I propose to document `autoconfigureRestBackupServer` field to mention which backup server will be used.

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+ARG KUBERNETES_VERSION=1.17.3
```
Comment (tolusha) [inline Dockerfile]: Is that kubernentes version related to che-operator somehow ?

File: .github/workflows/build-publish.yaml [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+      - name: Docker login
+        uses: azure/docker-login@v1
+        with:
+          login-server: quay.io
+          username: ${{ secrets.QUAY_USERNAME }}
+          password: ${{ secrets.QUAY_PASSWORD }}
+      - name: Docker Buildx
+        uses: crazy-max/ghaction-docker-buildx@v1.5.0
+        with:
+          version: v0.4.1
```
Comment (benoitf) [inline .github/workflows/build-publish.yaml]: please use the new location: https://github.com/crazy-max/ghaction-docker-buildx#moved-to-docker-organization

File: .github/workflows/pr-check.yaml [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    runs-on: ubuntu-20.04
+    steps:
+    - name: Clone source code
+      uses: actions/checkout@v1
+      with:
+        fetch-depth: 1
+    - name: Docker Buildx
+      uses: crazy-max/ghaction-docker-buildx@v1.4.0
+      with:
+        version: v0.4.1
```
Comment (benoitf) [inline .github/workflows/pr-check.yaml]: https://github.com/crazy-max/ghaction-docker-buildx#moved-to-docker-organization

File: Dockerfile (hunk 3 above)
Comment (benoitf) [inline Dockerfile]: we don't need ARCH as an argument

File: Dockerfile [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+ARG REST_SERVER_VERSION=0.10.0
+
+RUN apk add --no-cache curl && \
+    cd /usr/local/bin && \
+    curl -sLO https://storage.googleapis.com/kubernetes-release/release/v${KUBERNETES_VERSION}/bin/linux/${ARCH}/kubectl && \
```
Comment (benoitf) [inline Dockerfile:20]: you could use something like to grab `ARCH` https://github.com/eclipse-che/che-plugin-registry/blob/b5caed539d9faf117d4b8b73eabb8198653af9b1/sidecars/kubernetes-tooling/Dockerfile#L20-L22

File: rhel.Dockerfile [hunk 9]
```diff
@@ -0,0 +1,30 @@
+# Copyright (c) 2021 Red Hat, Inc.
+# This program and the accompanying materials are made
+# available under the terms of the Eclipse Public License 2.0
+# which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+
+# https://access.redhat.com/containers/?tab=tags#/registry.access.redhat.com/ubi8-minimal
+FROM registry.access.redhat.com/ubi8-minimal:8.1-409
```
Comment (benoitf) [inline rhel.Dockerfile]: should be `8.4-200`
```suggestion
FROM registry.access.redhat.com/ubi8-minimal:8.4-200
```

File: rhel.Dockerfile [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+
+ARG ARCH=amd64
```
Comment (benoitf) [inline rhel.Dockerfile]: we can automatically detect the arch using uname (so it works either with Docker build or Docker buildx)

File: README.md (hunk 4 above)
Comment (mmorhun) [reply README.md]: ok

File: .github/workflows/build-publish.yaml (hunk 6 above)
Comment (mmorhun) [reply .github/workflows/build-publish.yaml]: Thanks, will do

File: Dockerfile (hunk 3 above)
Comment (mmorhun) [reply Dockerfile]: Yes, we can autodetect it

File: Dockerfile (hunk 5 above)
Comment (mmorhun) [reply Dockerfile]: No, just a version. Updated to the latest stable

File: Dockerfile (hunk 3 above)
Comment (mmorhun) [reply Dockerfile]: Will do

**Review by tolusha — APPROVED** [review]

File: .github/workflows/build-publish.yaml [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+      - name: Clone source code
+        uses: actions/checkout@v1
+        with:
+          fetch-depth: 1
+      - name: Docker login
+        uses: azure/docker-login@v1
+        with:
+          login-server: quay.io
+          username: ${{ secrets.QUAY_USERNAME }}
+          password: ${{ secrets.QUAY_PASSWORD }}
```
Comment (benoitf) [inline .github/workflows/build-publish.yaml]: Sorry I didn't see it last time but here we should use docker action instead of azure
https://github.com/docker/login-action

File: Dockerfile [hunk 12]
```diff
@@ -0,0 +1,31 @@
+# Copyright (c) 2021 Red Hat, Inc.
+# This program and the accompanying materials are made
+# available under the terms of the Eclipse Public License 2.0
+# which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+
+FROM alpine:3.13.4
```
Comment (benoitf) [inline Dockerfile]:
```suggestion
FROM alpine:3.13.5
```

File: .github/workflows/pr-check.yaml [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+# This program and the accompanying materials are made
+# available under the terms of the Eclipse Public License 2.0
+# which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+name: PR check
+on:
+  pull_request:
+jobs:
```
Comment (benoitf) [inline .github/workflows/pr-check.yaml:12]: maybe here we should have a matrix to build as well the rhel.Dockerfile ? so we know that this alternate Dockerfile works as well

**Review by benoitf — APPROVED** [review]
few enhancements are welcome but it can be follow-up PRs
